pkg/sql/sem/tree/annotation.go

```diff
@@ -45,5 +45,8 @@ func (a *Annotations) Set(idx AnnotationIdx, annotation interface{}) {
 
 // Get an annotation from the container.
 func (a *Annotations) Get(idx AnnotationIdx) interface{} {
+	if len(*a) < int(idx) {
```

Comment:
There are circumstances (like within a UDF definition) where the annotations do not line up with the statement passed. This will cause panics, unless this check is added.